e2e/nextjs/src/app/api/app-directory-test/route.ts

```diff
@@ -1,25 +1,35 @@
-import { NextResponse } from 'next/server'
+import { NextRequest, NextResponse } from 'next/server'
 import { z } from 'zod'
+import { HighlightEnv, Highlight } from '@highlight-run/next/app-router'
+import CONSTANTS from '@/app/constants'
 
-export const GET = async function GET(request: Request) {
+const env: HighlightEnv = {
+	projectID: CONSTANTS.NEXT_PUBLIC_HIGHLIGHT_PROJECT_ID || '2',
+	otlpEndpoint: CONSTANTS.NEXT_PUBLIC_HIGHLIGHT_OTLP_ENDPOINT,
+	serviceName: 'vercel-app-directory',
+}
+
+const withHighlight = Highlight(env)
+
+export const GET = withHighlight(async function GET(request: NextRequest) {
```

Comment:
It turns out that we need to wrap App Router API endpoints after all 😥

Reply:
follow up feature, but what if our SDK could auto-detect and monkeypatch all serverless functions automatically? to avoid needing to do this wrapping, could our sdk auto-wrap?